internal/api/schemas/v1.yaml

```diff
@@ -271,6 +439,178 @@ components:
         privateKeyField:
           type: string
           minLength: 1
+    HTTPRoutePage:
+      type: object
+      required:
+      - routes
+      properties:
+        routes:
+          type: array
+          items:
+            "$ref": "#/components/schemas/HTTPRoute"
+        meta:
+          type: object
+    GatewayReference:
+      type: object
+      required:
+      - name
+      - namespace
+      properties:
+        name:
+          type: string
+          minLength: 1
+        namespace:
+          type: string
+          minLength: 1
+    HTTPRoute:
+      type: object
+      required:
+      - name
+      - namespace
+      - gateways
```

Comment:
Analogous to `ParentRefs`, K8s doesn't require these be set. Thoughts?

https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1alpha2.CommonRouteSpec

Reply:
I'd prefer to keep this as required, at least initially. I've never quite understood why a route doesn't need a gateway in the K8s API -- think maybe due to the initial thoughts around things like route delegation?

Either way, it'd be a breaking change to all of a sudden disallow empty gateways, but it's a lot easier to make the validation _less_ strict.

Reply:
Yea, spec flexibility will be useful here in the future for route delegation, GAMMA mesh binding, etc - requiring binding to a gateway for now makes sense to me.